storage/src/main/java/tech/pegasys/teku/storage/store/KeyValueStore.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.store;
+
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Generic simple key-value store interface Both key and value are not allowed to be null
+ *
+ * @param <TKey> key type
+ * @param <TValue> value type
+ */
+public interface KeyValueStore<TKey, TValue> {
+
+  /** Puts a new value. If the value is {@code null} then the entry is removed if exist */
```

Comment:
(nit) Might be more explicit to have a `del()` method instead of accepting `null` values.

Reply:
Yeah, though about it, but it makes batch update code a bit larger. But anyway the reason is rather my laziness than some rationality. Updated!